packages/template-retail-react-app/app/hooks/use-auth-modal.js

```diff
@@ -105,6 +105,8 @@ export const AuthModal = ({
 
         const handlePasswordlessLogin = async (email) => {
             try {
+                // Save the path where the user logged in
+                window.localStorage.setItem('returnToPage', window.location.pathname)
```

Comment:
What was the reason behind using localStorage? Can we use a query param to avoid having to handle this variable? Something like https://pwa-kit.storefront.com/login?token&1234&redirect_url=/checkout

Reply:
There are a few benefits of using query param than localStorage. One of which is it is not tied to a device. Imagine a shopper clicks login with link on a laptop, and then they check email and click on the magic link to logic. Can they resume their shopping from their?

Reply:
@alexvuong To send the `redirect_url` I would have to pass it in as part of the callbackURL param when calling the 1st `authorizePasswordless` API call so I thought this method would be more straightforward (though Blair did say that passing in the query params as part of the callback URL should be fine).

Reply:
@yunakim714 Why would we need to pass it into the callbackURL ? 🤔

Reply:
I'd suggest we try the query param solution since the localStorage can potential cause more bugs. There are some edge cases that will potentially disable the redirection.  Say, what if users clear their storage before they click on the magic link? or when they use incognito browser?